tests/unit/gesturehandler.cxx

```diff
@@ -0,0 +1,876 @@
+/* Copyright 2020 Pierre Ossman <ossman@cendio.se> for Cendio AB
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#include <stdio.h>
+#include <unistd.h>
+
+#include <vector>
+
+#include "../../vncviewer/GestureHandler.h"
+
+class TestClass : public GestureHandler
+{
+  protected:
+    virtual void handleGestureEvent(const GestureEvent& event);
+
+  public:
+    std::vector<GestureEvent> events;
+};
+
+void TestClass::handleGestureEvent(const GestureEvent& event)
+{
+  events.push_back(event);
+}
+
+#define ASSERT_TRUE(c) if (!(c)) { \
+  printf("FAILED on line %d (%s was false)\n", __LINE__, #c); \
+  return; \
+}
+// FIXME: handle doubles
+#define ASSERT_EQ(expr, val) if ((expr) != (val)) { \
+  printf("FAILED on line %d (%s equals %d, expected %d)\n", __LINE__, #expr, (int)(expr), (int)(val)); \
+  return; \
```

Comment:
Do we want to return on the first failure? Shouldn't we run all tests anyway?

Reply:
Yes, as that only returns from that specific test. There is no point continuing as that run is already broken.